src/utils/deferred.ts

```diff
@@ -50,7 +50,7 @@ export class Deferred<T> implements Promise<T> {
     this.#resolve(value);
   }
 
-  public reject(reason: any) {
+  public reject(reason: T) {
```

Comment:
reject is supposed to have a error, while resolve should have event message. So this line is not valid.

Reply:
Are you sure?

reject() calls this.#reason with the same `reason` argument, and the signature of reason is `#reject: (value: T) => void = () => {};`. So in practice, this is `T` as well.

Reply:
I assume you meant "`reject()` calls **`this.#reject`**". You are right, there is an issue, but it is in the `#reject` method. It should be `#reject: (value: unknown)`. I committed fix for it in this PR.

Reply:
Thanks, rebased